add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/App.kt

```diff
@@ -0,0 +1,13 @@
+package dev.flutter.multipleflutters
+
+import android.app.Application
+import io.flutter.embedding.engine.FlutterEngineGroup
+
+/**
+ * Application class for this app.
+ *
+ * This holds onto our engine group.
+ */
+class App : Application() {
+    val engines by lazy { FlutterEngineGroup(this) }
```

Comment:
don't know if we talked offline before or after this PR
 
```kotlin
lateinit val engines: FlutterEngineGroup

override fun onCreate() {
  super.onCreate()
  engines = ...
}
```

might solve the issue you had

Reply:
That would work, too.  I'm not sure if it is preferable to the lazy evaluation.  I googled it a bit and it isn't clear.  I can see some arguments either way.  I believe it's required just because you can't call abstract methods in the constructor, so literally any time after the constructor should be fine.  If that was not the case I think lateinit would be ideal since it would crash a bit earlier in a situation where we can't completely protect ourselves from crashing.

Reply:
Switch it, explained below.